packages/components/src/components/form/Range/Range.tsx

```diff
@@ -58,4 +49,76 @@ const Input = styled.input<Pick<RangeProps, 'thumbStyle' | 'trackStyle'>>`
     }
 `;
 
-export const Range = (props: RangeProps) => <Input type="range" {...props} />;
+const LabelsWrapper = styled.div<{ count: number; width?: number }>`
+    display: grid;
+    grid-template-columns: ${({ count, width }) =>
+        count && width && `repeat(${count}, ${width}px)`};
+    justify-content: space-between;
+`;
+
+const Label = styled.p<{ width?: number }>`
+    position: relative;
+    justify-self: center;
+    padding-top: 2px;
+    min-width: ${({ width }) => `${width}px`};
+    text-align: center;
+    font-size: ${variables.FONT_SIZE.TINY};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    opacity: 0.5;
+    cursor: pointer;
+`;
+
+export interface RangeProps {
+    className?: string;
+    max?: number;
+    min?: number;
+    step?: string;
+    onChange: React.ChangeEventHandler<HTMLInputElement>;
+    thumbStyle?: CSSObject;
+    trackStyle?: CSSObject;
+    value: number;
+    labels?: (string | number)[];
+    onLabelClick?: (value: number) => void;
+}
+
+export const Range = ({ labels, onLabelClick, ...props }: RangeProps) => {
+    const [laseElWidth, setLaseElWidth] = useState<number>();
+
+    const lastLabelRef = useRef<HTMLParagraphElement>(null);
+
+    useLayoutEffect(() => {
+        if (!lastLabelRef.current) {
+            return;
+        }
+
+        setLaseElWidth(lastLabelRef.current?.getBoundingClientRect().width);
+    }, [lastLabelRef, setLaseElWidth]);
+
+    const labelComponents = useMemo(
+        () =>
+            labels?.map((label, i) => {
+                const isLastElement = i === labels.length - 1;
+
+                return (
+                    <Label
+                        onClick={() => onLabelClick?.(Number.parseFloat(String(label)))}
+                        ref={isLastElement ? lastLabelRef : undefined}
+                    >
+                        {label}
+                    </Label>
+                );
+            }),
+        [labels, onLabelClick, lastLabelRef],
+    );
+
+    return (
+        <div>
+            <Input type="range" {...props} />
+            {labels?.length && (
+                <LabelsWrapper count={labels.length} width={laseElWidth}>
```

Comment:
`width` makes it to the DOM, should be a transient prop (`$width`) instead.

Reply:
44f52d3a252c5252508bf88c5c8c979357cd09e9

Reply:
44f52d3a252c5252508bf88c5c8c979357cd09e9